backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/dsl/impl/AbstractLuceneStandardIndexFieldTypeContext.java

```diff
@@ -60,7 +63,7 @@ public S projectable(Projectable projectable) {
 
 	@Override
 	public S indexNullAs(F indexNullAs) {
-		// TODO implement indexNullAs in the backends
+		indexNullAsValue = Optional.of( indexNullAs );
```

Comment:
`null` is not an acceptable value for `indexNullAs`. You can use `null` to represent "no value was selected", no need for an Optional.